tool/src/cpp2/ty.rs

```diff
@@ -537,8 +542,19 @@ impl<'ccx, 'tcx: 'ccx, 'header> TyGenContext<'ccx, 'tcx, 'header> {
                     expression: format!("{cpp_name}.AsFFI()").into(),
                 }]
             }
-            Type::Slice(hir::Slice::Str(..)) => {
-                // TODO: This needs to change if an abstraction other than std::string_view is used
+            Type::Slice(hir::Slice::Str(_, hir::StringEncoding::UnvalidatedUtf8)) => {
```

Comment:
nit: can we just merge these two branches?

Reply:
Wildcard is fine, all our string types are and will be some kind of slice with a pointer and a length